_includes/_components/domain-event.html

```diff
@@ -1,50 +1,133 @@
-<!-- TODO -->
-
-      <article class="component dark-blue">
-        <header>
-          <div class="container">
+<article class="component dark-blue">
+    <header>
+        <div class="container">
             <h2>Domain Event</h2>
-            <p>This component blablabla</p>
-          </div>
-        </header>
-        <div class="container content">
-          <p>
-            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
-          </p>
-          <div class="row">
+            <p>The raise method allows you to trigger any event on your entity and dispatched this Knp\Rad\DomainEvent\Event once the entity has been flushed.</p>
+        </div>
+    </header>
+    <div class="container content">
+        <div class="row">
             <div class="col-md-6">
-              <div class="standard">
-                <div class="banner"></div>
+                <div class="standard">
+                    <div class="banner"></div>
+{% highlight php %}
+<?php
+
+namespace App\Entity;
+
+class MyEntity
+{
+    // ...
+
+    public function setParam($value)
+    {
+        // ...
+    }
+}
+{% endhighlight %}
+                </div>
+
+                <div class="rad">
 {% highlight php %}
 <?php
 
-class Test
+namespace App\EventListener;
+
+use Doctrine\ORM\Event\PreUpdateEventArgs;
+use App\Entity\MyEntity;
+
+class MyEventListener
 {
-    public function test(SomeClass $subject)
+    public function preUpdate(PreUpdateEventArgs $event)
     {
-        return $subject->doSomething();
+        $entity = $event->getEntity();
+
+        if (!$entity instanceof MyEtity &&
+            !$event->hasChangedField('param'
+        )) {
+            return;
+        }
+
+        // Some stuff
     }
 }
 {% endhighlight %}
-              </div>
+                </div>
+
+                <div class="rad">
+{% highlight yaml %}
+# services.yml
+
+app.event_listener.my_event_listener:
+    class: AppBundle\EventListener\MyEventListener
+    tags:
+        - { name: doctrine.event_listener, event: preUpdate }
+{% endhighlight %}
+                </div>
             </div>
 
             <div class="col-md-6">
-              <div class="rad">
-                <div class="banner"></div>
+                <div class="rad">
+                    <div class="banner"></div>
 {% highlight php %}
 <?php
 
-class Test
+namespace App\Entity;
+
+use Knp\Rad\DomainEvent;
+
+class MyEntity implements DomainEvent\Provider
+{
+    // ...
```

Comment:
The entity has also to use a specific trait.

`use DomainEvent\ProviderTrait;`